packages/core/src/node/synchronizer.ts

```diff
@@ -332,6 +342,90 @@ export class Synchronizer extends EventEmitter {
       })
   }
 
+  async listenDepositUtxos(
+    addresses: ZkAddress[],
+    cb?: (utxo: UtxoSql) => void,
+  ) {
+    const lastDeposits = await this.db.findMany('Utxo', {
+      where: {},
+      orderBy: { depositedAt: 'desc' },
+      limit: 1,
+    })
+    const fromBlock = lastDeposits[0]?.blockNumber || 0
+    logger.info('found my deposit from block', fromBlock)
+    this.depositUtxoSubscriber = this.l1Contract.user.events
+      .DepositUtxo({
+        fromBlock,
+        filter: {
+          spendingPubKey: addresses.map(address => address.spendingPubKey()),
+        },
+      })
+      .on('connected', subId => {
+        logger.info(
+          `synchronizer.js: DepositUtxo listener is connected. Id: ${subId}`,
+        )
+      })
+      .on('data', async event => {
+        const { returnValues, blockNumber } = event
+        const owner = addresses.find(addr =>
+          Fp.from(returnValues.spendingPubKey).eq(addr.spendingPubKey()),
+        )
+        if (!owner) throw Error('Failed to find address')
+        const salt = Fp.from(returnValues.salt)
+        const note = new Note(owner, salt, {
+          eth: Fp.from(returnValues.eth),
+          tokenAddr: Fp.from(Address.from(returnValues.token).toBN()),
+          erc20Amount: Fp.from(returnValues.amount),
+          nft: Fp.from(returnValues.nft),
+        })
+        const utxo: UtxoSql = {
+          hash: note
+            .hash()
+            .toUint256()
+            .toString(),
+          eth: note
+            .eth()
+            .toUint256()
+            .toString(),
+          owner: owner.toString(),
+          salt: note.salt.toUint256().toString(),
+          tokenAddr: note
+            .tokenAddr()
+            .toUint256()
+            .toString(),
+          erc20Amount: note
+            .erc20Amount()
+            .toUint256()
+            .toString(),
+          nft: note
+            .nft()
+            .toUint256()
+            .toString(),
+          depositedAt: blockNumber,
+        }
+        logger.info(`synchronizer.js: Found My Deposit (${utxo.hash})`)
+        await this.blockCache.upsertCache(
+          'Utxo',
+          {
+            where: { hash: utxo.hash },
+            update: utxo,
+            create: utxo,
+          },
+          blockNumber,
+          event.blockHash,
+        )
+        if (cb) cb(utxo)
```

Comment:
why is there a callback in this function?

Reply:
Added a callback for the case we need any kind of post-processing for Deposit UTXOs. But we don't have any use cases for the callbacks now.

Reply:
Commit 173d222869abc14eaf95130067f598bcdec4ea97 resolves this